include/acquire.zarr.h

```diff
@@ -0,0 +1,274 @@
+#ifndef H_ACQUIRE_ZARR_V0
+#define H_ACQUIRE_ZARR_V0
+
+#include "zarr.types.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    typedef struct ZarrStreamSettings_s ZarrStreamSettings;
+    typedef struct ZarrStream_s ZarrStream;
+
+    /**
+     * @brief Get the version of the Zarr API.
+     * @return The version of the Zarr API.
+     */
+    uint32_t Zarr_get_api_version();
+
+    /**
+     * @brief Set the log level for the Zarr API.
+     * @param level The log level.
+     * @return ZarrStatus_Success on success, or an error code on failure.
+     */
+    ZarrStatus Zarr_set_log_level(ZarrLogLevel level);
+
+    /**
+     * @brief Get the log level for the Zarr API.
+     * @return The log level for the Zarr API.
+     */
+    ZarrLogLevel Zarr_get_log_level();
+
+    /**
+     * @brief Get the message for the given status code.
+     * @param status The status code.
+     * @return A human-readable status message.
+     */
+    const char* Zarr_get_error_message(ZarrStatus status);
+
+    /**
+     * @brief Create a Zarr stream settings struct.
+     * @return A pointer to the Zarr stream settings struct, or NULL on failure.
+     */
+    ZarrStreamSettings* ZarrStreamSettings_create();
+
+    /**
+     * @brief Destroy a Zarr stream settings struct.
+     * @details This function frees the memory allocated for the Zarr stream
+     * settings struct.
+     * @param[in] settings The Zarr stream settings struct.
+     */
+    void ZarrStreamSettings_destroy(ZarrStreamSettings* settings);
+
+    /**
+     * @brief Copy a Zarr stream settings struct.
+     * @param[in] settings The Zarr stream settings struct to copy.
+     * @return A copy of the Zarr stream settings struct.
+     */
+    ZarrStreamSettings* ZarrStreamSettings_copy(
```

Comment:
Is there a compelling reason to implement copying functionality for this object?

Reply:
Yes. If the settings are validated when the stream is created, and if the user retains the original data, then the stream should make a copy of the settings struct (or at least the data therein) so that the user can't inadvertently invalidate the stream by messing with the settings.